src/libraries/System.Memory.Data/src/System/BinaryData.cs

```diff
@@ -275,7 +275,8 @@ public override unsafe string ToString()
         [RequiresUnreferencedCode(JsonSerializerRequiresUnreferencedCode)]
         public T? ToObjectFromJson<T>(JsonSerializerOptions? options = default)
         {
-            return JsonSerializer.Deserialize<T>(_bytes.Span, options);
+            using var stream = new ReadOnlyMemoryStream(_bytes);
```

Comment:
We could also check for the preamble bytes and only use the `Stream` when necessary. Luckily we'd only need to check for the UTF-8 BOM as it is the only supported encoding.